content/docs/avalanche-l1s/upgrade/customize-avalanche-l1.mdx

```diff
@@ -758,6 +758,86 @@ With `allowFeeRecipients` enabled, validators can specify their addresses to col
 If `allowFeeRecipients` or `RewardManager` precompile is enabled on the Avalanche L1, but a validator doesn't specify a "feeRecipient", the fees will be burned in blocks it produces.
 </Callout>
 
+### Archival Node Configuration[​](#archival-node-configuration "Direct link to heading")
+
+Running an archival node that retains all historical state data requires specific configuration settings. Incorrect configuration can lead to historical data being pruned despite attempts to run in archival mode. Here are the key settings to configure:
+
+#### Configuration Directory Structure
+
+Avalanche node configurations are loaded from specific paths. For chain-specific configurations, use the following structure:
+
+```
+<chain-config-dir>/<blockchainID>/config.json
+```
+
+By default `<chain-config-dir>` is `~/.avalanchego/configs/chains/`. If you specify a custom directory with `--chain-config-dir`, remember that it should point to the parent directory of the chains folder, not include the "chains" part:
+
+```bash
+# CORRECT
+--chain-config-dir=/etc/.avalanchego/configs
+
+# INCORRECT
+--chain-config-dir=/etc/.avalanchego/configs/chains
+```
+
+If you want to use a different data directory overall, use the `--data-dir` flag:
+
+```bash
+--data-dir=/etc/.avalanchego/
+```
```

Comment:
I still don't understand why we need to explain these stuff as a part of archival setup. We should not direct people to use any extra custom flags unless very required. Otherwise we end-up having more confusions and misconfigurations. Can we move this section to p-chain flags (https://github.com/ava-labs/avalanchego/blob/master/config/config.md#data-directory)?